pkg/controller/argocd/sso.go

```diff
@@ -45,19 +93,66 @@ func (r *ReconcileArgoCD) reconcileSSO(cr *argoprojv1a1.ArgoCD) error {
 	if cr.Spec.SSO.Provider == argoprojv1a1.SSOProviderTypeKeycloak {
 		// TemplateAPI is available, Install keycloack using openshift templates.
 		if IsTemplateAPIAvailable() {
-			templateInstanceRef, err := newKeycloakTemplateInstance(cr.Namespace)
+			templateInstanceRef, err := newKeycloakTemplateInstance(cr)
 			if err != nil {
 				return err
 			}
 			existingTemplateInstance := template.TemplateInstance{}
-			err = r.client.Get(context.TODO(), types.NamespacedName{Name: templateInstanceRef.Name, Namespace: templateInstanceRef.Namespace}, &existingTemplateInstance)
+			err = r.client.Get(context.TODO(), types.NamespacedName{Name: templateInstanceRef.Name,
+				Namespace: templateInstanceRef.Namespace}, &existingTemplateInstance)
 			if err != nil && errors.IsNotFound(err) {
-				log.Info("Creating a new keycloak template instance")
+				log.Info(fmt.Sprintf("Template API found, Installing keycloak using openshift templates for ArgoCD %s in namespace %s",
+					cr.Name, cr.Namespace))
 				err = r.client.Create(context.TODO(), templateInstanceRef)
 				if err != nil {
 					return err
 				}
 			}
+
+			existingDC := &oappsv1.DeploymentConfig{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      defaultKeycloakIdentifier,
+					Namespace: cr.Namespace,
+				},
+			}
+			r.client.Get(context.TODO(), types.NamespacedName{Name: existingDC.Name, Namespace: existingDC.Namespace}, existingDC)
```

Comment:
We don't bother to error check?

Reply:
Template Instance takes some time to create the deployment config, Till then reconciller will not be able to find the Deployment Config.

Should we handle the NotFoundError as log message ?  something like 
`TemplateInstance not found or in the creation state`

Reply:
ack .. Thanks :)